library/entity/entity_events/src/main/java/org/quiltmc/qsl/entity_events/api/EntityKilledCallback.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021-2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.entity_events.api;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.Nullable;
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.base.api.event.EventAwareListener;
+
+/**
+ * A callback which is invoked on the logical server when an entity is killed by another entity.
+ *
+ * <p>The killed entity's {@link LivingEntity#getRecentDamageSource()} method can be used to retrieve the damage source
+ * which may have killed it.
+ */
+@FunctionalInterface
+public interface EntityKilledCallback extends EventAwareListener {
```

Comment:
Wouldn't this be more useful as a generic "Death" callback that might have an entity associated with it? The current javadoc implies this even should only be called when an entity is killed by another entity.

Reply:
agree, general death callback has more uses, and can be filtered by the listener to only listen for kills.

Reply:
agree, general death callback has more uses, and can be filtered by the listener to only listen for kills.

Reply:
I didn't make it a death callback because I was under the false assumption that multiple entity methods did not run super.onDeath(), making it difficult to maintanably inject an event invokation. However, from having a look it seems that only ServerPlayerEntity does this, and two injection points sounds reasonable.

Reply:
Done